app/models/compute_resources/foreman/model/libvirt.rb

```diff
@@ -151,12 +155,16 @@ def new_vm(attr = { })
       firmware_type = opts.delete(:firmware_type).to_s
       opts.merge!(process_firmware_attributes(opts[:firmware], firmware_type))
 
+      firmware_unprocesssed = opts.delete(:firmware_unprocessed).to_s
+      opts[:firmware] = firmware_unprocesssed unless create_vm_ctx || firmware_unprocesssed.empty?
```

Comment:
firmware_unprocessed is a bit unclear. Consider renaming it to something like unprocessed_firmware or initial_firmware for better readability.
```suggestion
      # initial_firmware is the firmware type assigned either by default or through the compute profile
      initial_firmware = opts.delete(:initial_firmware).to_s
      opts[:firmware] = initial_firmware if !create_vm_ctx && initial_firmware.present?

```